### DISCUSSION TITLE
[Enhancement] [drawer] Drawer 抽屉新增 `body-class` 属性

### DISCUSSION THREAD

**Discussion by WangJincheng4869** [element-plus/element-plus#19060] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `drawer`

## Description

目前虽然有 `class` 属性可用，但是当想修改 `body` 的样式时还是非常不方便的。
例如：
1. 自定义 body 的内边距
2. 设置禁用滚动条
3. 等其他自定义样式

如果支持 `body-class` 那么就可以使用类似 `tailwindcss` 这种框架来快速修改样式

### Additional comments

_(empty)_

**Comment by Dsaquel** [comment]
I mean you can always access to `.el-drawer__body` with `:deep(.el-drawer__body)` or having div containers like:

```vue
 <el-drawer v-model="drawer">
    
    <div class="body-container">content</div>
    
    <template #footer>
      <div class="footer-container">content</div>
    </template>

  </el-drawer>
 ```
 those for the workarounds.

But yeah that can be a feature request like `<table>` do with row-class-name, etc... but i dont know if its a good case for it.
Another solution is to create a sub component `<el-drawer-body>` but im not a fan of this too.

**Reply by WangJincheng4869** [reply]
这种方式的确可以使用，但是非常麻烦，所以才会有这个需求。

**Reply by Dsaquel** [reply]
Any thoughts @btea ?

**Reply by btea** [reply]
PRs are welcome.

**Comment by Dsaquel** [comment]
@WangJincheng4869 Here is a way to solve your problem correctly:

```css
/* global.css */
.my-custom-drawer-style {
  :deep(.el-drawer__header) {
    color: red
  }
  :deep(.el-drawer__body) {
    color: red
  }
}
```

```vue
//App.vue
import ./global.css
```

```vue
// random file
<template>
 <el-drawer v-model="drawer" class="my-custom-drawer-style">
    <template #header>
      this text is red
    </template>
this is also red
  </el-drawer>
</template>
```

**Reply by WangJincheng4869** [reply]
我知道这两种方式，目前也是这样做的，但是这样做非常麻烦。所以才会有这个需求。

**Reply by Dsaquel** [reply]
You'll still need :deep if we propose body-class in scoped style

**Reply by WangJincheng4869** [reply]
> You'll still need :deep if we propose body-class in scoped style

如果使用类似 `tailwindcss` 这类框架的话，可以不使用 `:deep` 的，因为这个 class 会直接对 card body 生效

**Reply by WangJincheng4869** [reply]
@Dsaquel 就像这样
![图片](https://github.com/user-attachments/assets/626d2670-20a9-4c2a-b8fe-f253cf64d499)

**Reply by Dsaquel** [reply]
@WangJincheng4869 working on it: [PR](https://github.com/element-plus/element-plus/pull/19099)

**Reply by Dsaquel** [reply]
@WangJincheng4869 resolved with `header-class`, `body-class` and `footer-class` => available in `2.9.3` [doc](https://element-plus.org/en-US/component/drawer.html#attributes)
